### ISSUE TITLE
[question] v3 - alertes osmose : faire/demander modif. FranceSpecificRules ?

### ISSUE THREAD

**Issue by deuzeffe** [osm-fr/osm-vs-fantoir#303]
Hello,

Puisque le code TOPO contient un caractère de moins que le code FANTOIR, tous les codes rentrés à partir de la v3 provoquent des alertes osmose (qu'il faut marquer en faux positif une par une :/)

Est-ce qu'il est possible que https://josm.openstreetmap.de/wiki/Rules/FranceSpecificRules, qu'osmose intègre dans ses analyses, soit facilement modifié dès à présent pour accepter les deux formats (qui vont coexister "un certain temps") ou c'est prématuré ?

**Comment by vdct** [comment]
C'est même carrément souhaitable oui. Tu saurais t'en charger ?

**Comment by deuzeffe** [comment]
Houlà, je ne peux me charger que d’ouvrir un ticket sur josm, vu que ce truc est bourré d'expressions rationnelles que je comprends encore moins que le langage overpass...

EDIT : done (https://josm.openstreetmap.de/ticket/23094#ticket)

**Comment by Olyon** [comment]
Il faudrait modifier les pages du wiki qui parlent de ref:FR:FANTOIR pour dire que la lettre de controle est optionnelle.
Il y a au moins la page https://wiki.openstreetmap.org/wiki/FR:Key:ref:FR:FANTOIR à modifier.

**Comment by deuzeffe** [comment]
@Olyon Just do it? ;)

**Comment by deuzeffe** [comment]
Pas l'impression que @PanierAvide a vu le ticket https://josm.openstreetmap.de/ticket/23094#ticket sur JOSM :(

**Comment by PanierAvide** [comment]
Pas vu passer le ticket, mais j'ai pas l'impression d'être à l'origine de la règle sur le FANTOIR, c'était plutôt les 4 premières règles du fichier... Il faut juste mettre en optionnel le 10e caractère du code c'est bien ça ?

**Comment by deuzeffe** [comment]
@PanierAvide oui, c'est ça. Me semble que @Olyon a répondu au ticket dans ce sens.

**Comment by deuzeffe** [comment]
@Olyon tu te sens d'intégrer la modif. que tu as proposée sur le code écrit par @didier2020 (si je lis bien l'historique du fichier) ?

**Comment by didier2020** [comment]
Bonjour,
si lettre de controle est optionnelle, il 'suffit" de modifier
([ABCDEFGHJKLMNPRSTUVWXYZ])
par
(|[ABCDEFGHJKLMNPRSTUVWXYZ])
dans le fichier des regles josm
https://josm.openstreetmap.de/wiki/Rules/FranceSpecificRules
(rien | [liste des lettres autorisées])

et mettre a jour la premiere ligne du wiki
"Le code FANTOIR a indiquer comme valeur doit faire 10 caractères..."

**Comment by deuzeffe** [comment]
@didier2020 je n'ai aucune connaissance pour savoir si ta solution est meilleure que celle d'@Olyon. je laisse qui sait, faire la modif. du fichier.

**Comment by frodrigo** [comment]
> @didier2020 je n'ai aucune connaissance pour savoir si ta solution est meilleure que celle d'@Olyon. je laisse qui sait, faire la modif. du fichier.

Les deux, et c'est déjà fait coté JOSM.

**Comment by deuzeffe** [comment]
@frodrigo vu, merci pour l'info et merci à @Olyon pour la modif.
@vdct on y est **enfin** arrivé.

**Comment by gosgtan** [comment]
Bonjour,
Malgré la modif, il y a toujours des alertes Osmose pour les FANTOIR à 9 caractères. Toutes les alertes que je me suis amusé à signaler en faux-positif le mois dernier sont revenues ce jour, et tous les FANTOIR que j'intègre nouvellement reviennent en alerte sur Osmose dès le lendemain également..

**Comment by gosgtan** [comment]
Bonjour,

Depuis aujourd'hui les FANTOIR à 10 caractères renvoient une alerte sur Osmose... Il aurait plutôt fallu pouvoir faire cohabiter ceux à 9 et 10 caractères, non? Tout modifier pour enlever le dernier caractère ça va être chronophage

**Comment by deuzeffe** [comment]
@gosgtan Modification d'aujourd'hui par @didier2020 (merci)
https://josm.openstreetmap.de/wiki/Rules/FranceSpecificRules?action=diff&version=92
Ça devrait encore s'améliorer.

**Comment by frodrigo** [comment]
Le changement n'a été déployé que hier soir, mais avec un bug. Correction et redéploiement aujourd’hui. Ça devrait être bon demain.

**Comment by deuzeffe** [comment]
@frodrigo merci pour les détails !

**Comment by gosgtan** [comment]
Impeccable merci !
